packages/hardhat-core/src/internal/hardhat-network/provider/fork/rpcToTxData.ts

```diff
@@ -1,34 +1,33 @@
 import { AccessListEIP2930TxData, TxData } from "@ethereumjs/tx";
 
+import { FeeMarketEIP1559TxData } from "@ethereumjs/tx/dist/types";
 import { RpcTransaction } from "../../../core/jsonrpc/types/output/transaction";
 
 export function rpcToTxData(
   rpcTransaction: RpcTransaction
-): TxData | AccessListEIP2930TxData {
-  const txData: AccessListEIP2930TxData = {
+): TxData | AccessListEIP2930TxData | FeeMarketEIP1559TxData {
+  return {
     gasLimit: rpcTransaction.gas,
-    gasPrice: rpcTransaction.gasPrice,
+    // NOTE: RPC EIP-1559 txs still have this field for backwards compatibility,
+    //  but FeeMarketEIP1559TxData doesn't.
+    gasPrice:
+      rpcTransaction.type?.eqn(2) ?? false
```

Comment:
I can't create a suggestion for this, but I would extract this to a variable:

```ts
  const isEIP1559 = rpcTransaction.type?.eqn(2) ?? false;

  return {
    gasLimit: rpcTransaction.gas,
    // NOTE: RPC EIP-1559 txs still have this field for backwards compatibility,
    //  but FeeMarketEIP1559TxData doesn't.
    gasPrice: isEIP1559 ? undefined : rpcTransaction.gasPrice,
```

Reply:
Done